src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkService.java

```diff
@@ -23,11 +23,19 @@
    */
   void startTask(String tableName, String topic, int partition);
 
+  /**
+   * call pipe to insert a collections of JSON records
+   * will trigger time based flush
+   * @param records record content
+   */
+  void insert(final Collection<SinkRecord> records);
+
   /**
    * call pipe to insert a JSON record
+   * will not trigger time based flush
    * @param record record content
    */
-  void insert(SinkRecord record);
+  void insert(final SinkRecord record);
```

Comment:
Kept two insert method exposed in the interface, the first one will trigger time based flush, the second one won't. Did this for testing.